File: javascript/tests/palindrome_number_test.js
import {assert, assertFalse} from "@std/assert";
import {isPalindrome} from "@/palindrome_number.js";


Deno.test(function exampleCases() {
    assert(isPalindrome(121));
    assertFalse(isPalindrome(-121));
    assertFalse(isPalindrome(10));
});

Deno.test(function extraCases() {
    assert(isPalindrome(1221));
    assert(isPalindrome(0));
    assert(isPalindrome(-0));
});

File: java/src/test/java/com/jamesansley/PalindromeNumberTest.java
package com.jamesansley;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

public class PalindromeNumberTest {
    @Test
    public void exampleCases() {
        PalindromeNumber solution = new PalindromeNumber();

        assertTrue(solution.isPalindrome(121));
        assertFalse(solution.isPalindrome(-121));
        assertFalse(solution.isPalindrome(10));
    }

    @Test
    public void extraCases() {
        PalindromeNumber solution = new PalindromeNumber();

        assertTrue(solution.isPalindrome(1221));
        assertTrue(solution.isPalindrome(0));
        assertTrue(solution.isPalindrome(-0));
    }
}
